config/m4/ib.m4

```diff
@@ -279,6 +288,20 @@ AS_IF([test "x$with_ib" == xyes],
                              [], [#include <infiniband/verbs_exp.h>])
            ])
 
+       # Device Memory support
+       AS_IF([test "x$with_dm" != xno],
+           [AC_TRY_COMPILE([#include <infiniband/verbs_exp.h>],
+               [
+                   void* a1 = ibv_exp_alloc_dm;
+                   void* a2 = ibv_exp_reg_mr;
```

Comment:
why need to check reg_mr and dereg_mr?

Reply:
check all used calls

Reply:
imo, looks excessive.

Reply:
agree, but this API is declared as "inline" and classic macro for API detection could not be used (it is based on linkage), that is why I used func-ptr trick to detect existing inline calls